sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/OutputWriter.scala

```diff
@@ -57,6 +57,9 @@ abstract class OutputWriterFactory extends Serializable {
  * executor side.  This instance is used to persist rows to this single output file.
  */
 abstract class OutputWriter {
+  /** Initializes before writing any rows. Invoked on executor size. */
+  def init(): Unit
```

Comment:
Rather than make subclasses implement as a no-op, just provide that no-op impl here?

Reply:
yeah makes sense, i will make that change